client/src/containers/CampaignAdmin/CampaignAdmin.js

```diff
@@ -387,114 +290,145 @@ By removing this item all linked items will be removed unless used elsewhere.`;
     }
 
     return <Custom key={props.name} {...props} />;
-  }
+  };
 
-  addCampaign() {
-    const path = this.getActionRoute(0, 'create');
-    this.props.router.navigate(path);
-  }
+  const addCampaign = () => {
+    const path = getActionRoute(0, 'create');
+    router.navigate(path);
+  };
+
+  const prevPropsRef = useRef();
```

Comment:
The only ref before was the help button one. I can see this is used in the `useEffect` that replaces `componentDidUpdate()` which means refactoring was probably done. Can you please explain the changes and why a new `if (prevPropsRef.current) {` condition was added to that?

Kinda feels like it's being used instead of a dependencies array for the use effect hook.